### PR TITLE
feat(tenants): adds support for Create and Read Tenants

### PR DESCRIPTION
**Pull request by cristhian-castaneda**
## Description
Update `armory config get` &  `armory config apply --file <path to file>` commands that you can use to Create or Read tenants configuration information for a specified organization. Delete/Update is not included in this PR.

`armory config get` output

```yaml
tenants:
  - armory-cloud
  - armory-cloud-dev
roles:
  - name: Demo-Role
    tenant: armory-cloud-dev
    grants:
      - type: api
        resource: deployment
        permission: full
```

## Motivation and Context
https://armory.atlassian.net/browse/CDAAS-1735

### To Read Tenants
- Run command `armory config get`
- Verify the output contains the `tenants` list

### To Create Tenants
- Run command `armory config get`  (Save the output in a file called `configuration.yaml`)
- Update `configuration.yaml` adding a new entry (tenant name) under the `tenants` property
- Run command `armory config apply -f  configuration.yaml`
- Run command `armory config get`
- Verify the output contains the new tenant you added

# Have your performed the following tests?

Please confirm you have done the following tests to ensure your pull request is ready to be merged..

- [ ] If this change modifies the deployment flow have triggered a deployment using this branch (i.e. `build/dist/darwin_amd64/armory deploy start -f <path to file>`)?
- [ ] If this change modifes the login flow have you tried logging in and out with this branch (i.e. `build/dist/darwin_amd64/armory login`) and verified the credentials work?
- [ ] Have you verified the specific change made in this PR?

### CODE REVIEW

**Review by Greg-At-Armory — APPROVED** [review]

File: pkg/configuration/service.go [hunk 1]
```diff
@@ -29,3 +29,10 @@ func CreateRoleRequest(config *model.RoleConfig) (*configClient.CreateRoleReques
 	}
 	return &req, nil
 }
+
+func CreateEnvironmentRequest(environment string) (configClient.CreateEnvironmentRequest, error) {
```
Comment (danielpeach) [inline pkg/configuration/service.go]: You don't have to follow the pattern of the other functions here - don't make the function return an `error`

File: cmd/config/get_test.go [hunk 2]
```diff
@@ -72,7 +78,45 @@ func (suite *ConfigGetTestSuite) TestConfigGetUserRole() {
...
+	}}
+	var getRolesExpected []model.RoleConfig
+
+	if err := registerResponder(getEnvironmentsExpected, http.StatusOK, "/environments", http.MethodGet); err != nil {
+		suite.T().Fatal(err)
+	}
+
+	err := registerResponder(getRolesExpected, http.StatusOK, "/roles", http.MethodGet)
+	if err != nil {
+		suite.T().Fatalf("TestDeployStartJsonSuccess failed with: %s", err)
```
Comment (danielpeach) [inline cmd/config/get_test.go]: I'm not sure why every log line in this file says "TestDeployStartJsonSuccess", but you certainly don't have to follow that pattern 😄 

If you call Fatalf, I'm pretty sure the `testing` package gives you pretty good clues about which test called it - you don't need to include the name of the test in the log.

Comment (danielpeach) [reply cmd/config/get_test.go]: In my opinion you should not `Fatalf` here at all. 

In most cases I'd just do `assert.NoError(t, registerResponder(getRolesExpected, http.StatusOK, "/roles", http.MethodGet))`

See https://github.com/golang/go/wiki/TestComments#keep-going

File: cmd/config/get.go [hunk 3]
```diff
@@ -3,16 +3,19 @@ package config
 import (
 	"context"
 	"fmt"
+	_nethttp "net/http"
```
Comment (danielpeach) [inline cmd/config/get.go]: I know you didn't write this but I have no idea why this is renamed `_nethttp` 🤷 

There might have been a good reason at some point but I don't see it now. If you wouldn't mind cleaning this up, that would be great. Always good to leave things a little cleaner than you found them.

**Review by danielpeach — APPROVED** [review]

File: cmd/config/get_test.go (hunk 2 above)
Comment (cristhian-castaneda) [reply cmd/config/get_test.go]: Thanks @danielpeach this is pretty useful, I just clean up most that I could, that was a very good reading

**Pull request merged**